### DISCUSSION TITLE
Make `io.strimzi.operator.common.model.Labels` model public

### DISCUSSION THREAD

**Discussion by mikekamornikov** [strimzi/strimzi-kafka-operator#7983] [Ideas]
Hello,
I'd like to use `Labels` functionality to build `KafkaConnector` and `KafkaTopic` objects. So far as I see it's not possible to avoid hardcoding Strimzi lables. So I wonder if you can start releasing `operator-common` or as an alternative inject functionality into Builder classes or even move it completely to some public module like `api`.

**Comment by scholzj** [comment]
What do you want to do with the Labels class? It has nothing to do with the `api` module - you cannot really use it in the `api` classes. The whole `operator-common` module is not public API in any way and the `Labels` class is not really part of any stable API either. So it would be good to better understand what pain points you have and what might be the solutions.

**Reply by mikekamornikov** [reply]
Actually I just need the names of Strimzi constants to do something like that while creating Strimzi CRs in my app:
```java
new KafkaConnectorBuilder(new KafkaConnector())
        .editOrNewMetadata()
            .addToLabels(Labels.STRIMZI_CLUSTER_LABEL, "cluster-name")
        .endMetadata()
```
or even this way (as I have a template to modify):
```java
new KafkaConnectorBuilder(connectorTemplate)
        .editOrNewMetadata()
            .addToLabels(Labels.fromResource(connectorTemplate)
                .withStrimziCluster("kafka-cluster")
                .toMap())
        .endMetadata()
```
As you understand right now I have to hardcode label name (`addToLabels("strimzi.io/cluster", "cluster-name")`). Obviously it's a minor thing, but looks a little bit dirty anyway.

**Reply by scholzj** [reply]
Ok, I think that makes sense. I do not think we should move the `Labels` class -> but the constants like this should be somewhere in the `api` module. I opened #7988 to cover it.
